hikari/impl/rest_bot.py

```diff
@@ -0,0 +1,499 @@
+# -*- coding: utf-8 -*-
+# cython: language_level=3
+# Copyright (c) 2020 Nekokatt
+# Copyright (c) 2021 davfsa
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Standard implementations of a Interaction based REST-only bot."""
+from __future__ import annotations
+
+__all__: typing.Sequence[str] = ["RESTBot"]
+
+import asyncio
+import typing
+
+from hikari import config
+from hikari import traits
+from hikari.api import interaction_server as interaction_server_
+from hikari.impl import entity_factory as entity_factory_impl
+from hikari.impl import interaction_server as interaction_server_impl
+from hikari.impl import rest as rest_impl
+from hikari.internal import ux
+
+if typing.TYPE_CHECKING:
+    import concurrent.futures
+    import socket as socket_
+    import ssl
+
+    from hikari import applications
+    from hikari.api import entity_factory as entity_factory_
+    from hikari.api import rest as rest_
+    from hikari.api import special_endpoints
+    from hikari.interactions import bases as interaction_bases
+
+
+class RESTBot(traits.InteractionServerAware, interaction_server_.InteractionServer):
```

Comment:
I feel like `RESTBot` will cause confusion with `RESTApp` :confused: 

Maybe?
```suggestion
class InteractionBot(traits.InteractionServerAware, interaction_server_.InteractionServer):
```

Reply:
Why would people think that RESTApp is a bot?

Reply:
also random point, would it be worth it renaming "Bot" to something like "GatewayBot"?

Reply:
That rename would just cause more naming issues than fixing them

Reply:
wdym fixing them